src/components/Board.js

```diff
@@ -5,21 +5,81 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+
+const CARDS_URL ="https://inspiration-board.herokuapp.com/boards/Cara/cards/";
+
 
 class Board extends Component {
   constructor() {
     super();
 
-    this.state = {
-      cards: [],
+    this.state ={
+       cards:[]
+
     };
   }
 
+  componentDidMount() {
+    axios.get(CARDS_URL)
+    .then((response)=> {
+      this.setState({cards: response.data});
+    })
+    .catch((error) =>{
+        this.setState({error: error.message});
+    });
+  }
+
+  removeCard = (id) =>{
+    // console.log(cardKey,'cardKey')
+    axios.delete(CARDS_URL+id)
+    .then((response) => {
+      let updatedData = this.state.cards;
+      this.setState({cards: updatedData});
+      this.componentDidMount();
```

Comment:
I haven't seen specific advice to avoid this, but I'm pretty certain that our code should never be directly calling a lifecycle method such as `componentDidMount`.

If you want your callback in `removeCard` to do the same thing, then code from within `componentDidMount` should be moved in a separate method (e.g. `loadCards`) and that method should be called from both places:

```javascript
loadCards() {
  axios.get(CARDS_URL)
  .then(...)
  .catch(...);
}

componentDidMount() {
  this.loadCards();
}

removeCard = (id) => {
  axios.delete(CARDS_URL+id)
  .then((response) => {
    this.loadCards();
  })
  .catch(...);
}
```

Reply:
Additionally, the way this delete callback is implemented ultimately requires two requests to the server: one for the delete, and a second to request all of the card data gain.

An alternative approach (not always the right option, there are trade-offs for both ways), would be to have your callback send the delete requests and then also directly remove the card data from state (similar to how the `addCard` callback directly adds a card into the state).